test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -335,82 +191,70 @@ protected Settings restClientSettings() {
-            String testPath = testCandidate.getSuitePath() + "/" + testCandidate.getTestSection().getName();
-            assumeFalse("[" + testCandidate.getTestPath() + "] skipped, reason: blacklisted", blacklistedPathMatcher
-                    .isSuffixMatch(testPath));
-        }
-        //The client needs non static info to get initialized, therefore it can't be initialized in the before class
-        restTestExecutionContext.initClient(clusterUrls, restClientSettings());
-        restTestExecutionContext.clear();
-
-        //skip test if the whole suite (yaml file) is disabled
-        assumeFalse(buildSkipMessage(testCandidate.getSuitePath(), testCandidate.getSetupSection().getSkipSection()),
-                testCandidate.getSetupSection().getSkipSection().skip(restTestExecutionContext.esVersion()));
-        //skip test if the whole suite (yaml file) is disabled
-        assumeFalse(buildSkipMessage(testCandidate.getSuitePath(), testCandidate.getTeardownSection().getSkipSection()),
-            testCandidate.getTeardownSection().getSkipSection().skip(restTestExecutionContext.esVersion()));
-        //skip test if test section is disabled
-        assumeFalse(buildSkipMessage(testCandidate.getTestPath(), testCandidate.getTestSection().getSkipSection()),
-                testCandidate.getTestSection().getSkipSection().skip(restTestExecutionContext.esVersion()));
-    }
-
-    private static String buildSkipMessage(String description, SkipSection skipSection) {
-        StringBuilder messageBuilder = new StringBuilder();
-        if (skipSection.isVersionCheck()) {
-            messageBuilder.append("[").append(description).append("] skipped, reason: [").append(skipSection.getReason()).append("] ");
-        } else {
-            messageBuilder.append("[").append(description).append("] skipped, reason: features ")
-                    .append(skipSection.getFeatures()).append(" not supported");
-        }
-        return messageBuilder.toString();
+    /**
+     * Get the list of hosts in the cluster.
+     */
+    protected final List<HttpHost> getClusterHosts() {
+        return clusterHosts;
     }
 
-    public void test() throws IOException {
-        //let's check that there is something to run, otherwise there might be a problem with the test section
-        if (testCandidate.getTestSection().getExecutableSections().size() == 0) {
-            throw new IllegalArgumentException("No executable sections loaded for [" + testCandidate.getTestPath() + "]");
+    private RestClient buildClient(Settings settings) throws IOException {
+        HttpHost[] hosts = new HttpHost[clusterHosts.size()];
+        {
+            int i = 0;
+            for (HttpHost host : clusterHosts) {
+                // Allow configuring HTTPS in the settings
+                hosts[i++] = new HttpHost(host.getHostName(), host.getPort(), settings.get(PROTOCOL, "http"));
```

Comment:
We can't read the protocol when creating the hosts the first time around because the protocol comes from the settings which are actually per client. I originally changed it so we could created the hosts one time. I can change it back to that if you think it makes more sense. Honestly, I like it better with the `getProtocol` method so we only build the things once but if we want to use the settings we have to do it this way.